src/components/ReportActionItem/MoneyRequestView.tsx

```diff
@@ -349,10 +349,18 @@ function MoneyRequestView({
     const shouldShowNotesViolations = !isReceiptBeingScanned && canUseViolations && ReportUtils.isPaidGroupPolicy(report);
     const shouldShowReceiptHeader = isReceiptAllowed && (shouldShowReceiptEmptyState || hasReceipt);
 
-    const errors = {
-        ...(transaction?.errorFields?.route ?? transaction?.errors),
-        ...parentReportAction?.errors,
-    };
+    const errors = useMemo(() => {
+        const combinedErrors = {
+            ...(transaction?.errorFields?.route ?? transaction?.errors),
+            ...parentReportAction?.errors,
+        };
+        return Object.fromEntries(
+            Object.entries(combinedErrors).map(([key, value]) =>
+                // swap for translation for each error message
+                [key, swapForTranslation(value as string)],
+            ),
+        );
```

Comment:
This is kind of a new pattern, so I want to check with the team first to see if everyone is cool with it, as recommended by our checklist.

I don't really like that we're doing the swapping within the component. Ideally the swapping would be done when saving an Onyx update from the network or pusher. Let's pause this work until we hear what the team thinks. Sorry for the delay, but if we build this right it will be useful in the future instead of a headache.

Reply:
@dominictb yeah, please revert the `swapForTranslation` changes. I think we'll just deal with the slightly strange non-translated error message.